### PR TITLE
Adding the kafka broker SRC image for future usage

### PR DESCRIPTION
**Pull request by matzew**
Signed-off-by: Matthias Wessendorf <mwessend@redhat.com>

Similar to eventing/kafka, adding the broker src-code to this repo 

/assign @mgencur

### CODE REVIEW

**Review by pierDipi — APPROVED** [review]
/lgtm

File: openshift/ci-operator/source-image/Dockerfile [hunk 1]
```diff
@@ -4,5 +4,6 @@ COPY oc /usr/bin/oc
 COPY --from=registry.ci.openshift.org/openshift/knative-v1.0.1:knative-serving-src /go/src/knative.dev/serving/ /go/src/knative.dev/serving/
 COPY --from=registry.ci.openshift.org/openshift/knative-v1.0.0:knative-eventing-src /go/src/knative.dev/eventing/ /go/src/knative.dev/eventing/
 COPY --from=registry.ci.openshift.org/openshift/knative-v1.0.0:knative-eventing-kafka-src /go/src/knative.dev/eventing-kafka/ /go/src/knative.dev/eventing-kafka/
+COPY --from=registry.ci.openshift.org/openshift/knative-v1.0.0:knative-eventing-kafka-broker-src /go/src/knative.dev/eventing-kafka-broker/ /go/src/knative.dev/eventing-kafka-broker/
 
 RUN chmod g+w /go/src/knative.dev/serving/ && chmod g+w /go/src/knative.dev/eventing/ && chmod g+w /go/src/knative.dev/eventing-kafka/
```
Comment (mgencur) [inline openshift/ci-operator/source-image/Dockerfile:9]:
```suggestion
RUN chmod g+w /go/src/knative.dev/serving/ && chmod g+w /go/src/knative.dev/eventing/ && chmod g+w /go/src/knative.dev/eventing-kafka/ && chmod g+w /go/src/knative.dev/eventing-kafka-broker/
```

File: hack/generate/dockerfile.sh [hunk 2]
```diff
@@ -16,6 +16,7 @@ values[VERSION]="$(metadata.get project.version)"
 values[SERVING_VERSION]="$(metadata.get dependencies.serving)"
 values[EVENTING_VERSION]="$(metadata.get dependencies.eventing)"
 values[EVENTING_KAFKA_VERSION]="$(metadata.get dependencies.eventing_kafka)"
+values[KNATIVE_EVENTING_KAFKA_BROKER_VERSION]="$(metadata.get dependencies.eventing_kafka_broker)"
```
Comment (mgencur) [inline hack/generate/dockerfile.sh:19]:
```suggestion
values[EVENTING_KAFKA_BROKER_VERSION]="$(metadata.get dependencies.eventing_kafka_broker)"
```

Comment (mgencur) [reply hack/generate/dockerfile.sh:19]: This is just to unify the approach. The other variables do not have the "KNATIVE" prefix either.

File: templates/test-source-image.Dockerfile [hunk 3]
```diff
@@ -4,5 +4,6 @@ COPY oc /usr/bin/oc
 COPY --from=registry.ci.openshift.org/openshift/knative-v__SERVING_VERSION__:knative-serving-src /go/src/knative.dev/serving/ /go/src/knative.dev/serving/
 COPY --from=registry.ci.openshift.org/openshift/knative-v__EVENTING_VERSION__:knative-eventing-src /go/src/knative.dev/eventing/ /go/src/knative.dev/eventing/
 COPY --from=registry.ci.openshift.org/openshift/knative-v__EVENTING_KAFKA_VERSION__:knative-eventing-kafka-src /go/src/knative.dev/eventing-kafka/ /go/src/knative.dev/eventing-kafka/
+COPY --from=registry.ci.openshift.org/openshift/knative-v__KNATIVE_EVENTING_KAFKA_BROKER_VERSION__:knative-eventing-kafka-broker-src /go/src/knative.dev/eventing-kafka-broker/ /go/src/knative.dev/eventing-kafka-broker/
```
Comment (mgencur) [inline templates/test-source-image.Dockerfile:7]:
```suggestion
COPY --from=registry.ci.openshift.org/openshift/knative-v__EVENTING_KAFKA_BROKER_VERSION__:knative-eventing-kafka-broker-src /go/src/knative.dev/eventing-kafka-broker/ /go/src/knative.dev/eventing-kafka-broker/
```

**Review by mgencur — CHANGES_REQUESTED** [review]

File: templates/test-source-image.Dockerfile [hunk 4]
```diff
@@ -4,5 +4,6 @@ COPY oc /usr/bin/oc
...
 
 RUN chmod g+w /go/src/knative.dev/serving/ && chmod g+w /go/src/knative.dev/eventing/ && chmod g+w /go/src/knative.dev/eventing-kafka/
```
Comment (mgencur) [inline templates/test-source-image.Dockerfile:9]:
```suggestion
RUN chmod g+w /go/src/knative.dev/serving/ && chmod g+w /go/src/knative.dev/eventing/ && chmod g+w /go/src/knative.dev/eventing-kafka/ && chmod g+w /go/src/knative.dev/eventing-kafka-broker/
```

File: openshift/ci-operator/source-image/Dockerfile (hunk 1 above)
Comment (mgencur) [reply openshift/ci-operator/source-image/Dockerfile:9]: This is actually not necessary. We only need to change it in the template below and re-generate.

Comment (matzew) [reply openshift/ci-operator/source-image/Dockerfile:9]: I think that was changed, when doing the generate ? not sure what you mean

**Comment by matzew** [comment]
@mgencur thanks for the suggestions - I applied them, and rebased this PR :heart:

**Comment by matzew** [comment]
/retest
:thinking:

**Pull request merged**